pt/orm/retrieving-data-and-resultsets.rst

```diff
@@ -1,17 +1,1220 @@
-Requisitando dados e conjuntos de resultados
-############################################
+Retornando dados e conjuntos de resultados
+##########################################
+
+.. php:namespace:: Cake\ORM
+
+.. php:class:: Table
+
+Quando executar uma query, você obterá um objeto Entidade. Nesta sessão discutiremos diferentes caminhos para se obter: entidades, carregar informações relacionais, abstratas, ou complexo relacional. Você poderá ler mais sobre :doc:`/orm/entities` ( ‘Entity’ em inglês ).
+
+Depurando Queries e Resultados
+==============================
+
+Quando o ORM foi implementado, era muito difícil depurar os resultados obtidos nas versões anteriores do CakePHP. Agora existem muitas formas fáceis de inspecionar os dados retornados pelo ORM.
+
+- ``debug($query)`` Mostra o SQL e os parâmetros incluídos, não mostra resultados.
+- ``debug($query->all())`` Mostra a propriedade ResultSet retornado pelo ORM.
+- ``debug($query->toArray())`` Um caminho mais fácil para mostrar todos os resultados.
+- ``debug(json_encode($query, JSON_PRETTY_PRINT))`` Exemplo em JSON.
+- ``debug($query->first())`` Primeiro resultado obtido na query.
+- ``debug((string)$query->first())`` Mostra as propriedades de uma única entidade em JSON.
+
+Tente isto na camada Controller: debug( $this->{EntidadeNome}->find()->all() );
+
+Pegando uma entidade com a chave primária
+=========================================
+
+.. php:method:: get($id, $options = [])
+
+Sempre que é necessário editar ou visualizar uma entidade ou dados relacionais você pode usar ``get()``::
+
+    // No controller ou table tente isto.
+
+    // Retorna um único artigo pelo id primário.
+    $article = $articles->get($id);
+
+    // Retorna um artigo com seus comentários
+    $article = $articles->get($id, [
+        'contain' => ['Comments']
+    ]);
+
+Quando não conseguir obter um resultado
+``Cake\Datasource\Exception\RecordNotFoundException`` será disparado. Você poderá tratar esta exceção, ou converter num erro 404.
+
+O metódo ``find()`` usa uma cache integrado. Você pode uma a opção ``cache`` quando chamar ``get()`` para uma performance na leitura - ``caching``::
+
+    // No controller ou table tente isto.
+
+    // Use uma configuração de cache ou uma instância do CacheEngine do Cake com uma ID gerada.
+    $article = $articles->get($id, [
+        'cache' => 'custom',
+    ]);
+
+    // mykey reserva uma id especifica para determinado cache de resultados.
+    $article = $articles->get($id, [
+        'cache' => 'custom', 'key' => 'mykey'
+    ]);
+
+    // Desabilitando cache explicitamente
+    $article = $articles->get($id, [
+        'cache' => false
+    ]);
+
+
+
+
+``Por padrão o CakePHP possui um sistema interno de cache que viabiliza busca e aumenta a performance - não recomendado desabilatar.``
```

Comment:
yes, in portuguese language, some code explained other than English. Sorry for my patience.